3rdparty/stout/tests/decompress_tests.cpp

```diff
@@ -0,0 +1,318 @@
+  //       22       22 2018-02-21 10:06 hello   Howdy there, partner!\n
+  // --------  -------                  ------- ------
+  //       22       22                  1 file
+
+  ASSERT_SOME(os::write(path.get(), base64::decode(
+      "H4sICE61jVoAA2hlbGxvLnRhcgDtzjEOwjAQRNGtOcXSU9hx7FyBa0RgK0IR"
+      "RsYIcfsEaGhQqggh/VfsFLPFDHEcs6zLzEJon2k7bz7zrQliXdM6Nx/vxVgb"
+      "uiBqVt71crvWvqjKKaZ0yOnr31L/p/b5fnxoHWKJO730pZ5j2W5+vQoAAAAA"
+      "AAAAAAAAAAAAsGQC2DPIjgAoAAA=").get()));
+
+  // Note: The file does NOT have a .tar.gz extension. We could rename
+  // it, but libarchive doesn't care about extensions. It determines
+  // the format from the contents of the file. So this is tested here
+  // as well.
+  EXPECT_SOME(decompress::extract(path.get(), ""));
+
+  string extractedFile = path::join(os::getcwd(), "hello");
+  ASSERT_TRUE(os::exists(extractedFile));
+
+  ASSERT_SOME_EQ("Howdy there, partner!\n", os::read(extractedFile));
+}
+
+
+TEST_F(DecompressTest, DecompressTarFile)
+{
+  // Construct a hello.tar file that can be decompressed
+  string dir = path::join(os::getcwd(), "somedir");
+  ASSERT_SOME(os::mkdir(dir));
+
+  Try<string> path = os::mktemp(path::join(dir, "XXXXXX"));
+  ASSERT_SOME(path);
+
+  // The .tar file, since not compressed, is long. So go through some
+  // pains to construct the contents fo the file programmatically.
+  //
+  // We could skip the .tar file test, but it's worth having it.
+
+  //  Length     Size     Date    Time  Name    Content
+  // --------  ------- ---------- ----- ----    ------
+  //    10240    10240 2018-02-21 10:06 hello   Howdy there, partner (.tar)!\n
+  // --------  -------                  ------- ------
+  //    10240    10240                  1 file
+
+  string tarContents =
+      "aGVsbG8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAADAwMDA2NjQAMDAwMTc1MAAwMDAxNzUwADAwMDAwMDAwMDM1"
+      "ADEzMjQ1MTA2NTE1ADAxMTY3NAAgMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB1c3RhciAgAGplZmZj"
+      "b2YAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAamVmZmNvZgAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      "AAAAAAAAAAAAAAAAAAAAAABIb3dkeSB0aGVyZSwgcGFydG5lciEgKC50YXIp"
+      "CgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+  for (int i = 0; i < 214; i++)
+  {
+      tarContents +=
+          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+  }
+
+  tarContents += "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==";
+
+  // Now write out the .tar file, compress it, and verify results
```

Comment:
"decompress" but actually not, we're just extracting an uncompressed tar

Reply:
Fixed.